### ISSUE TITLE
Some questions about single view reconstruction

### ISSUE THREAD

**Issue by wzxshhz123** [mabaorui/NeuralPull#4]
Hi, mabaorui

Recently I want to re-Implement the single view reconstruction part of your impressive work with PyTorch, but I am not sure about how to train network without point cloud(which is very important for loss computation and optimization). Maybe I should use the image and its corresponding point cloud at the same time and use point cloud to sample query points just like point cloud reconstruction?

Besides, I found you load 2D features directly from local but with dir name 'AtlasNetown' rather than SoftRes related. Is that mean I should train Atlastnet firstly to extract image features?

Best wishes,
wzxshhz123

**Comment by mabaorui** [comment]
Thanks for your dedication.
When training a single-view reconstruction network, first run 'python sample_query_point.py' to obtain training data for loss computation, and then input image to extract features.
We leverage the 2D encoder used by SoftRas to infer the 2D image conditions.So you can use the Pre-training encoder to extract image features and then input to NeuralPull (in my code is to extract it in advance and save it to an npz file), or you can also use the network structure of SoftRas to train from scratch .
[1]Liu, S., Li, T., Chen, W., and Li, H. Soft rasterizer: A differentiable renderer for image-based 3D reasoning. IEEE
International Conference on Computer Vision, 2019a.
